### PR TITLE
[Homepage] Promos and Hero

### PR DESCRIPTION
**Pull request by JaeBea**
## Ticket(s)
**Jira Ticket**
- [EWL-6309: Homepage - promo incorrectly styled](https://issues.ama-assn.org/browse/EWL-6309)
- [EWL-6307: Homepage - hero has an unwanted line](https://issues.ama-assn.org/browse/EWL-6307)

## Description
Adds missing flex properties to flexbox children and adjusts widths and margins so that promo box ratios match Zeplin and do not overlap text. Removes left border in homepage hero on D8. 

### Dependencies
[Homepage: Hero and Articles #911](https://github.com/AmericanMedicalAssociation/ama-d8/pull/911)

## To Test
- Pull `bugfix/homepage-hero-promo` branch
- Run `gulp serve` in ama-style-guide-2/styleguide to ensure a local version of style guide is running.
- This work relies on template updates in [Homepage: Hero and Articles #911](https://github.com/AmericanMedicalAssociation/ama-d8/pull/911). Update ama-d8 branch and local provision vars accordingly. 
- Ensure you have prod content before testing further.
- Go to the default homepage and confirm hero no longer has vertical line separating left and right sections.
- Confirm promos have proper ratios and do not have overlapping text.
- Test this in IE 11, Safari, and Firefox to ensure flexbox widths work correctly.

## Visual Regressions
A report is available [here](http://htmlpreview.github.io/?https://github.com/AmericanMedicalAssociation/ama-style-guide-2/blob/visual-regression-testing-artifact/bugfix/homepage-hero-promo/html_report/index.html).

## Relevant Screenshots/GIFs
Home page promos in D8
image

Original Incorrect Section
image

## Remaining Tasks
N/A

## Additional Notes
N/A

### CODE REVIEW

File: styleguide/source/assets/scss/05-pages/_homepage.scss [hunk 1]
```diff
@@ -3,6 +3,9 @@
   .ama__layout--one-column__content-top {
     display: none;
   }
+  .ama__layout--split__right{
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/05-pages/_homepage.scss]:
```suggestion
  .ama__layout--split__right {
```

File: styleguide/source/assets/scss/02-molecules/_partner-promo.scss [hunk 2]
```diff
@@ -44,5 +46,10 @@
     &__promo-heading{
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/02-molecules/_partner-promo.scss]:
```suggestion
    &__promo-heading {
```

File: styleguide/source/assets/scss/02-molecules/_partner-promo.scss [hunk 3]
```diff
@@ -23,10 +23,12 @@
     color: $gray-64;
 
     // Right now the style guide uses `buttons`, but eventually we want to switch these to `a` elements.
-    button, a {
+    button, a, .ama__button {
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/02-molecules/_partner-promo.scss]:
```suggestion
    a, 
    button, 
    .ama__button {
```

File: styleguide/source/assets/scss/05-pages/_homepage.scss (hunk 1 above)
Comment (nstriedinger) [inline styleguide/source/assets/scss/05-pages/_homepage.scss]:
```suggestion
  .ama__layout--split__right {
```

**Review by nstriedinger — CHANGES_REQUESTED** [review]

**Comment by JaeBea** [comment]
@nstriedinger Code spacing has been updated as you suggested.

**Pull request merged**